lib/https.js

```diff
@@ -40,6 +40,12 @@ function Server(opts, requestListener) {
   }
   opts = util._extend({}, opts);
 
+  if (opts && typeof opts.key === 'boolean')
+    throw new Error('"options.key" must not be a boolean');
+
+  if (opts && typeof opts.cert === 'boolean')
+    throw new Error('"options.cert" must not be a boolean');
```

Comment:
By the way, we try to migrate to a [new error system](https://github.com/nodejs/node/blob/master/doc/guides/using-internal-errors.md) that allows for more flexibility in the future, it would be cool if you think you could take a look at using that.